### PR TITLE
Field overrides: Filter by field name using regex

### LINKED ISSUE #26831: Field Overrides: Multi-Field Matching
**What would you like to be added**:
When using Overrides, The "Matcher > Filter by field" only lets you select a single field. We need at least a regex matcher to apply overrides to multiple fields. (Numeric / String / Time field matchers might be useful as well, but a regex would move us forward I think)

**Why is this needed**:
A response can have many series (e.g. United States, group by State). Additionally, the data may be very dynamic so the fields are not know ahead of time. 

Now that we are returning  data with Frames, users depend on overrides to get their display right in certain circumstances.

### PR DESCRIPTION
**Pull request by dprokop**
Closes https://github.com/grafana/grafana/issues/26831

Introduces a new matcher UI that enables matching fields for overrides using a regular expression.

@kylebrandt @mckn do you think this ultra-simple UI is fine for the needs?

----
**tldr;**
As an alternative, we could introduce a sort of combo UI that would either allow selecting from the list or providing a regex, similar as we do with Filter by name transformation. My initial idea was to have that, but I found it too complex:

Kapture 2020-08-19 at 13 34 36

### CODE REVIEW

**Comment by kylebrandt** [comment]
Also see https://github.com/grafana/grafana/pull/27056

**Comment by dprokop** [comment]
> Also see #27056

@kylebrandt thanks for the heads up. I guess these two are complementary and might be useful :)

**Comment by oddlittlebird** [comment]
Docs look good.

**Comment by ryantxu** [comment]
> do you think this ultra-simple UI is fine for the needs?

Initially yes -- a small improvement would be to indicate how many fields (if any!) match the pattern.  may also be good to add autocomplete for regexp exact match?

**Comment by ryantxu** [comment]
I removed * from the type matcher, since it feels better to do it explicitly in something that says regex :)

**Review by ryantxu — APPROVED** [review]

**Comment by ryantxu** [comment]
@dprokop -- to be clear, I think this is perfect as is for 7.1.4, but we could give it more love for 7.2+
